pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -654,7 +654,7 @@ func getTraitName(ac *v1alpha2.ApplicationConfiguration, componentName string,
 	}
 
 	if len(traitName) == 0 {
-		traitName = util.GenTraitName(componentName, ct.DeepCopy())
+		traitName = util.GenTraitName(componentName, ct.DeepCopy(), kind)
```

Comment:
why use kind? shouldn't we use traitdefinition name?

Reply:
In pure OAM Kubernetes Runtime, the traitdefinition name is normally very long, like `autoscalers.standard.oam.dev`, but for both runtime and KubeVela, the Kind is short and can stand for the traitdefinition name. 

How about I cutting the first part split by `.` as the trait type in runtime? In Kubevela the name is short and clear.

Reply:
cutting the first part to be the name is fine for me.
Please make sure the naming rule is clearly documented somewhere.